epochStart/metachain/legacySystemSCs.go

```diff
@@ -762,14 +778,21 @@ func (s *legacySystemSCProcessor) replaceValidators(
 	old state.ValidatorInfoHandler,
 	new state.ValidatorInfoHandler,
 	validatorsInfoMap state.ShardValidatorsInfoMapHandler,
-) error {
-	stakingV4Enabled := s.enableEpochsHandler.IsFlagEnabled(common.StakingV4StartedFlag)
-	if stakingV4Enabled {
-		return validatorsInfoMap.Replace(old, new)
+) {
+	// legacy code
+	if !s.enableEpochsHandler.IsFlagEnabled(common.StakingV4StartedFlag) {
+		_ = validatorsInfoMap.ReplaceValidatorByKey(old.GetPublicKey(), new, old.GetShardId())
+		return
 	}
 
-	_ = validatorsInfoMap.ReplaceValidatorByKey(old.GetPublicKey(), new, old.GetShardId())
-	return nil
+	// try with new code which does extra validity checks.
```

Comment:
this should not "try" something... are the new checks used in other places? how is the error handling there? if any new check fails and we revert to the old... we could just do the old...

Reply:
`.Replace` is used in 2 other places (after stakingV4 is enabled), where we correctly call the functiona and return error if exists, since that is new code and I tried to correctly use the func